dice_ml/data_interfaces/public_data_interface.py

```diff
@@ -356,6 +356,18 @@ def prepare_query_instance(self, query_instance, encode):
 
             return temp.tail(test.shape[0]).reset_index(drop=True)
 
+    def prepare_dataset_instance(self, encode=True):
+        """Prepares the entire dataset as a sequence of test inputs for DiCE."""
+        dataset_dict = self.data_df[self.feature_names].to_dict('records')

```

Comment:
this is a bit odd. Line 361 converts df to dict, and then 362 converts it back to dict. Is there a better way to do this?
One way will be to break up the prepare query instance function into two, and only call the second part.